frontend/src/components/authentication/ForgotPassword.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {useState, useCallback} from 'react'
+import {CssBaseline, TextField} from '@material-ui/core'
+import {Button, Avatar, Typography, Link, Paper, Grid, Box }  from '@material-ui/core'
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined'
+
+import * as gridSize from '../../constants/styles.values'
+import {SIX as SHADOW_DEPTH_SIX} from '../../constants/styles.values'
+import { emailFormatIsValid } from '../../utils/FormatVerifications'
+import {useStyles} from './SignUp.styles'
+import { AuthApiClient } from '../../api/authRequest'
+
+export default function ForgotPasswordSide(): React.ReactElement {
+
+  const classes = useStyles()
+
+  const initialState = {
+    email: '',
+    message: '',
+    error: false
+  }
+
+  const [state, setState] = useState(initialState)
+  
+  const onChangeEmail = useCallback((e) => {
+    setState({...state, email: e.target.value})
+  }, [setState, state])
+  
+  const recoverPassword = useCallback((e) => {
+    e.preventDefault()
+    if (emailFormatIsValid(state.email)) {
+      const authApiClient = new AuthApiClient()
+      authApiClient.recover(state.email).then((res) => {
+        setState({...state, message: res.message, error: res.error})
```

Comment:
I think we can use separate states:
```
const [email, setEmail] = useState()
const [serverResponseMsg, setServerResponseMsg] = useState()
const [error, setError] = useState()
// so you can make "setState" bit more readable
const {message, error} = response
setServerResponseMsg(messge)
setError(error)
```

Reply:
I did it how it's normally done in react components. Separate states make more lines of code and more dependencies in functions.
But ok, if you like it like that, I'll change the code.